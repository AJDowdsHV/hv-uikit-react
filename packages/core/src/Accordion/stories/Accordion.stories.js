import React, { useState, useMemo } from "react";
import { makeStyles } from "@mui/styles";
import {
  HvAccordion,
  HvListContainer,
  HvListItem,
  HvInput,
  HvButton,
  HvTable,
  HvTableHead,
  HvTableHeader,
  HvTableRow,
  HvTableBody,
  HvTableCell,
  useHvData,
  useHvTableSticky,
} from "../..";

export default {
  title: "Display/Accordion",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvAccordion } from '@hitachivantara/uikit-react-core'",
  },

  component: HvAccordion,
};

export const Main = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="item1" label="Analytics" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item3" label="Data" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

export const Disabled = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="disabled-item1" label="Analytics" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item3" label="Data" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

Disabled.parameters = {
  docs: {
    description: { story: "A common accordion with disable items. " },
  },
};

export const Controlled = () => {
  const useStyles = makeStyles((theme) => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
    buttonContainer: {
      "& button": {
        marginRight: theme.hvSpacing("xs"),
        marginBottom: theme.hvSpacing("md"),
      },
    },
  }));
  const classes = useStyles();
  const [expandedState, setExpandedState] = useState({ analytics: true, system: true, data: true });
  const handleToggle = (key) => {
    const newValue = { ...expandedState };
    newValue[key] = !newValue[key];
    setExpandedState(newValue);
  };
  const handleAll = (option) => {
    setExpandedState({ analytics: option, system: option, data: option });
  };
  return (
    <>
      <div className={classes.buttonContainer}>
        <HvButton category="secondary" onClick={() => handleToggle("analytics")}>
          Toggle Analytics
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("system")}>
          Toggle System
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("data")}>
          Toggle Data
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(false)}>
          Close all
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(true)}>
          Expand all
        </HvButton>
      </div>
      <div className={classes.accordionContainer}>
        <HvAccordion
          id="controlled-item1"
          label="Analytics"
          onChange={() => handleToggle("analytics")}
          expanded={expandedState.analytics}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Views</HvListItem>
            <HvListItem>Parameters</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item2"
          label="System"
          onChange={() => handleToggle("system")}
          expanded={expandedState.system}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Settings</HvListItem>
            <HvListItem>Network</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item3"
          label="Data"
          onChange={() => handleToggle("data")}
          expanded={expandedState.data}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Storage</HvListItem>
            <HvListItem>Memory</HvListItem>
          </HvListContainer>
        </HvAccordion>
      </div>
    </>
  );
};

Controlled.parameters = {
  docs: {
    description: { story: "An accordion controlled externally forcing expansion." },
  },
};

export const Form = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      maxWidth: 300,
    },
    formContainer: {
      padding: "0 32px",
      "& > div": {
        paddingTop: 17,
      },
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="personal" label="Personal Information" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-name" label="Name" placeholder="Insert first name" required />
          <HvInput id="input-email" label="Email" placeholder="Insert your email" required />
          <HvInput id="input-phone" label="Phone" placeholder="Insert your phone number" />
          <HvInput id="input-extension" label="Extension" placeholder="Insert phone extension" />
          <HvInput id="input-country" label="Country" placeholder="Insert country name" />
          <HvInput id="input-province" label="City/Province" placeholder="Insert province name" />
        </div>
      </HvAccordion>
      <HvAccordion id="billing" label="Billing Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city" label="City" placeholder="Insert city name" />
          <HvInput id="input-state" label="State" placeholder="Insert state" />
          <HvInput id="input-code" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
      <HvAccordion id="shipping" label="Shipping Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address-bill" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2-bill" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city-bill" label="City" placeholder="Insert city name" />
          <HvInput id="input-state-bill" label="State" placeholder="Insert state" />
          <HvInput id="input-code-bill" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
    </div>
  );
};

Form.parameters = {
  docs: {
    description: { story: "An accordion composition with forms." },
  },
};

export const Typography = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      width: "100%",
    },
  }));
  const classes = useStyles();

  const getData = () => [
    {
      id: 1,
      title: "A new hope",
      director: "George Lucas",
      producer: "Gary Kurzt",
      release: "1977-05-25",
    },
    {
      id: 2,
      title: "Empires strike back",
      director: "Irvin Kershner",
      producer: "Gary Kurzt",
      release: "1980-05-17",
    },
    {
      id: 3,
      title: "Return of the jedi",
      director: "Richard Marquand",
      producer: "Howard Kazanjian",
      release: "1983-05-25",
    },
  ];

  const getColumns = () => [
    {
      Header: "Title",
      accessor: "title",
      sticky: "left",
      width: 150,
      style: { textTransform: "capitalize" },
    },
    {
      Header: "Director",
      accessor: "director",
    },
    {
      Header: "Producer",
      accessor: "producer",
    },
    {
      Header: "Release Date",
      accessor: "release",
    },
  ];

  const data = useMemo(() => getData(), []);
  const columns = useMemo(() => getColumns(), []);

  const { getTableProps, getTableHeadProps, getTableBodyProps, prepareRow, headerGroups, rows } =
    useHvData(
      {
        columns,
        data,
        stickyHeader: true,
      },
      useHvTableSticky
    );

  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="film" label="Films" headingLevel={2}>
        <HvTable id="accordion-table" {...getTableProps()}>
          <HvTableHead {...getTableHeadProps()}>
            {headerGroups.map((headerGroup) => (
              <HvTableRow {...headerGroup.getHeaderGroupProps()}>
                {headerGroup.headers.map((col) => (
                  <HvTableHeader {...col.getHeaderProps()}>{col.render("Header")}</HvTableHeader>
                ))}
              </HvTableRow>
            ))}
          </HvTableHead>
          <HvTableBody {...getTableBodyProps()}>
            {rows.map((row) => {
              prepareRow(row);

              return (
                <HvTableRow {...row.getRowProps()}>
                  {row.cells.map((cell) => (
                    <HvTableCell {...cell.getCellProps()}>{cell.render("Cell")}</HvTableCell>
                  ))}
                </HvTableRow>
              );
            })}
          </HvTableBody>
        </HvTable>
      </HvAccordion>
    </div>
  );
};

Typography.parameters = {
  docs: {
    description: { story: "An accordion with a different button typography." },
  },
};
