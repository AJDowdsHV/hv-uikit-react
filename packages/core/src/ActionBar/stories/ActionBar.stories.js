import React from "react";
import { makeStyles } from "@material-ui/core";
import { HvActionContainer, HvButton, HvDropDownMenu } from "../..";

export default {
  title: "Patterns/Action Bar/Action Container",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionContainer } from '@hv/uikit-react-core/dist'"
  },
  component: HvActionContainer
};

export const Main = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

export const DualAction = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
      </HvActionContainer>
    </div>
  );
};

DualAction.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};

export const VariedActionBar = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvDropDownMenu
              onClick={(e, item) => console.log(item.label)}
              dataList={[{ label: "Delete" }, { label: "Update" }]}
            />
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

VariedActionBar.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};
