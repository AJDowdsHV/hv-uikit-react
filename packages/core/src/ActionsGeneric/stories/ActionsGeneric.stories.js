/* eslint-disable react/prop-types */
import React from "react";
import { Add, Delete, Lock, Preview } from "@hv/uikit-react-icons";
import HvActionsGeneric from "..";

export default {
  title: "Components/Actions Generic",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionsGeneric } from '@hv/uikit-react-core/dist'",
    dsVersion: "3.2.0",
  },
  component: HvActionsGeneric,
  decorators: [
    (Story) => (
      <div
        style={{
          display: "flex",
          justifyContent: "center",
        }}
      >
        <Story />
      </div>
    ),
  ],
};

export const Main = () => (
  <HvActionsGeneric
    actions={[
      { id: "add", label: "Add", iconCallback: () => <Add /> },
      { id: "delete", label: "Delete", iconCallback: () => <Delete /> },
      { id: "lock", label: "Lock", iconCallback: () => <Lock /> },
      { id: "put", label: "Preview", iconCallback: () => <Preview /> },
    ]}
    maxVisibleActions={2}
  />
);
