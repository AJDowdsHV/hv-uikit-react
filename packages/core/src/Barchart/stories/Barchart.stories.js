import { makeStyles } from "@material-ui/core";
import withStyles from "@material-ui/core/styles/withStyles";
import React, { useState, useEffect } from "react";
import { Random } from "../../utils";
import { HvBarchart, HvDropDownMenu, HvDropdown, HvTypography } from "../..";

export default {
  title: "Visualizations/Bar Chart",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvBarchart } from '@hv/uikit-react-core/dist'"
  },
  component: HvBarchart
};

export const Main = () => {
  const layout = {
    yaxis: { showline: false }
  };
  return (
    <HvBarchart
      layout={layout}
      data={[
        {
          x: ["January", "February", "March"],
          y: [2300, 1000, 8500],
          name: "Sales Target"
        }
      ]}
    />
  );
};

export const VerticalWithSingleTooltip = () => {
  const trace1 = {
    x: ["January", "February", "March"],
    y: [2300, 1000, 8500],
    name: "Sales Target"
  };

  const data = [trace1];

  const layout = {
    yaxis: { showline: false }
  };

  return <HvBarchart data={data} tooltipType="single" layout={layout} />;
};

VerticalWithSingleTooltip.parameters = {
  docs: {
    description: { story: "Tooltip in single line mode." }
  }
};

export const GroupedVerticalBarchart = () => {
  const data = [
    { x: ["Group 1", "Group 2", "Group 3"], y: [2300, 1000, 8500], name: "Sales Target" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [6000, 3900, 1000], name: "Sales Per Rep" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [3700, 7500, 1100], name: "Monthly Sales" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [2100, 8500, 3000], name: "Target" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [500, 8000, 9500], name: "Cash" }
  ];

  const layout = {
    legend: { orientation: "h", x: 0.2, y: 1.1 },
    yaxis: { showline: false }
  };

  return <HvBarchart data={data} layout={layout} />;
};

GroupedVerticalBarchart.parameters = {
  docs: {
    description: { story: "Representation of groups by using multiple bars." }
  }
};

export const CustomStackedVerticalBarchart = () => {
  const styles = () => ({
    wrapper: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center"
    }
  });

  const ChartHeader = withStyles(styles)(({ classes, children }) => (
    <div className={classes.wrapper}>{children}</div>
  ));

  const data = [
    {
      x: ["ASR3", "HAL9", "DR21", "HY54", "KW65", "RE98", "ZX52", "UI56"],
      y: [50, 350, 420, 310, 390, 420, 200, 430],
      name: "Uploads"
    },
    {
      x: ["ASR3", "HAL9", "DR21", "HY54", "KW65", "RE98", "ZX52", "UI56"],
      y: [370, 80, 60, 280, 310, 320, 110, 190],
      name: "Downloads"
    }
  ];

  const layout = {
    legend: { orientation: "h", x: 0.4, y: 1.1 },

    yaxis: {
      ticksuffix: " Gb",
      showline: false
    }
  };

  const useStyles = makeStyles(() => ({
    titlePadding: { marginTop: 10 },
    dropdownPlacement: {
      marginTop: 24,
      marginLeft: 10
    },
    controllerGroup: {
      display: "flex"
    }
  }));

  const classes = useStyles();

  return (
    <>
      <ChartHeader>
        <HvTypography className={classes.titlePadding} variant="mTitle">
          Server Status Summary
        </HvTypography>
        <div className={classes.controllerGroup}>
          <HvDropdown
            id="dropdown2"
            labels={{ title: "Time Period" }}
            classes={{ root: classes.root }}
            values={[
              { label: "Last 0.5h" },
              { label: "Last 1.5h", selected: true },
              { label: "Last 24h" },
              { label: "Last 48h" }
            ]}
          />
          <HvDropDownMenu
            className={classes.dropdownPlacement}
            onClick={(e, item) => console.log(item.label)}
            dataList={[{ label: "Label 1" }, { label: "Label 2" }, { label: "Label 3" }]}
          />
        </div>
      </ChartHeader>

      <HvBarchart stack data={data} layout={layout} />
    </>
  );
};

CustomStackedVerticalBarchart.story = {
  parameters: {
    docs: {
      storyDescription: "Groups in stack mode."
    }
  }
};

export const StackedVerticalBarchart = () => {
  const data = [
    { x: ["Group 1", "Group 2", "Group 3"], y: [2300, 1000, 8500], name: "Sales Target" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [6000, 3900, 1000], name: "Sales Per Rep" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [3700, 7500, 1100], name: "Monthly Sales" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [2100, 8500, 3000], name: "Target" },
    { x: ["Group 1", "Group 2", "Group 3"], y: [500, 8000, 9500], name: "Cash" }
  ];

  const layout = {
    legend: { orientation: "h", x: 0.2, y: 1.1 },
    yaxis: { showline: false }
  };

  return <HvBarchart stack data={data} layout={layout} />;
};

StackedVerticalBarchart.parameters = {
  docs: {
    description: { story: "Groups in stack mode." }
  }
};

export const SimpleHorizontalBarchart = () => {
  const layout = {
    legend: { orientation: "h", x: 0.2, y: 1.1 },
    xaxis: { showline: false }
  };

  return (
    <HvBarchart
      horizontal
      layout={layout}
      data={[
        {
          x: [2300, 1000, 8500],
          y: ["January", "February", "March"],
          name: "Sales Target"
        }
      ]}
    />
  );
};

SimpleHorizontalBarchart.parameters = {
  docs: {
    description: { story: "Representation of horizontal chart." }
  }
};

export const HorizontalBarchartWithSingleTooltip = () => {
  const layout = {
    legend: { orientation: "h", x: 0.2, y: 1.1 },
    xaxis: { showline: false }
  };

  return (
    <HvBarchart
      horizontal
      tooltipType="single"
      data={[
        {
          x: [2300, 1000, 8500],
          y: ["January", "February", "March"],
          name: "Sales Target"
        }
      ]}
      layout={layout}
    />
  );
};

HorizontalBarchartWithSingleTooltip.parameters = {
  docs: {
    description: { story: "Tooltip in single line mode." }
  }
};

export const GroupedHorizontalBarchart = () => {
  const data = [
    { y: ["Group 1", "Group 2", "Group 3"], x: [2300, 1000, 8500], name: "Sales Target" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [6000, 3900, 1000], name: "Sales Per Rep" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [3700, 7500, 1100], name: "Monthly Sales" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [2100, 8500, 3000], name: "Target" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [500, 8000, 9500], name: "Cash" }
  ];

  const layout = {
    legend: { orientation: "h", x: 0.2, y: 1.1 },
    xaxis: { showline: false }
  };

  return <HvBarchart horizontal data={data} layout={layout} />;
};

GroupedHorizontalBarchart.parameters = {
  docs: {
    description: { story: "Representation of groups by using multiple horizontal bars." }
  }
};

export const StackedHorizontalBarchart = () => {
  const data = [
    { y: ["Group 1", "Group 2", "Group 3"], x: [2300, 1000, 8500], name: "Sales Target" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [6000, 3900, 1000], name: "Sales Per Rep" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [3700, 7500, 1100], name: "Monthly Sales" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [2100, 8500, 3000], name: "Target" },
    { y: ["Group 1", "Group 2", "Group 3"], x: [500, 8000, 9500], name: "Cash" }
  ];
  const layout = {
    legend: { orientation: "h", x: 0.2, y: 1.1 },
    xaxis: { showline: false }
  };

  return <HvBarchart stack data={data} horizontal layout={layout} />;
};

StackedHorizontalBarchart.parameters = {
  docs: {
    description: { story: "Groups in stack mode." }
  }
};

export const WithIntervalUpdates = () => {
  const r = new Random();

  const [data, setData] = useState([
    {
      x: ["January", "February", "March"],
      y: [2300, 1000, 8500],
      name: "Sales Target"
    }
  ]);

  useEffect(() => {
    const interval = setTimeout(() => {
      setData([
        {
          x: ["January", "February", "March"],
          y: [r.next(1000, 3000), r.next(500, 3500), 8500],
          name: "Sales Target"
        }
      ]);
    }, 2000);

    return () => clearTimeout(interval);
  });

  const layout = {
    yaxis: { showline: false }
  };

  return <HvBarchart data={data} layout={layout} />;
};

WithIntervalUpdates.parameters = {
  docs: {
    description: { story: "Data updated each second." }
  }
};
