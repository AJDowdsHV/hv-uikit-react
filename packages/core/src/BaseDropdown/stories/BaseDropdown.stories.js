import React, { useState } from "react";
import { MoreOptionsVertical } from "@hv/uikit-react-icons";
import { useTheme } from "@material-ui/core";
import HvBaseDropdown from "..";
import HvInput from "../../Input";
import HvButton from "../../Button";
import HvTypography from "../../Typography";
import { HvFormElement, HvInfoMessage, HvLabel } from "../../Forms";

export default {
  title: "Components/BaseDropdown",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvBaseDropdown } from '@hv/uikit-react-core/dist'"
  },
  component: HvBaseDropdown
};

export const Main = () => <HvBaseDropdown placeholder="Placeholder..." />;

Main.story = {
  parameters: {
    v3: true,
    pa11y: {
      ignore: ["region"]
    }
  }
};

export const WithContent = () => {
  const theme = useTheme();
  return (
    <HvBaseDropdown
      placeholder={<HvTypography id="labelTypography">Placeholder</HvTypography>}
      aria-labelledby="labelTypography"
    >
      <div style={{ width: "300px", height: "300px", background: theme.palette.atmo2 }}>
        <HvInput id="input-dropdown" />
      </div>
    </HvBaseDropdown>
  );
};

WithContent.story = {
  parameters: {
    v3: true
  }
};

export const WithForms = () => (
  <HvFormElement id="withForms" disabled={false}>
    <HvLabel style={{ paddingBottom: "8px", display: "block" }} label="Label" />
    <HvBaseDropdown placeholder="Placeholder..." />
    <HvInfoMessage disableGutter style={{ paddingTop: "8px", display: "block" }}>
      The description.
    </HvInfoMessage>
  </HvFormElement>
);

WithForms.story = {
  parameters: {
    v3: true
  }
};

export const Controlled = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <HvButton onClick={() => setOpen(!open)}>change</HvButton>
      <p />
      <HvBaseDropdown
        placeholder="PlaceHolder..."
        expanded={open}
        onToggle={(e, s) => setOpen(s)}
      />
    </>
  );
};

Controlled.story = {
  parameters: {
    v3: true
  }
};

export const Custom = () => {
  const [open, setOpen] = useState(false);

  const theme = useTheme();

  const setterOpen = () => setOpen(!open);

  return (
    <HvBaseDropdown
      expanded={open}
      onClickOutside={setterOpen}
      component={
        <HvButton
          overrideIconColors={false}
          onClick={setterOpen}
          category="icon"
          aria-label="Dropdown menu"
          style={{ backgroundColor: open ? theme.palette.atmo1 : undefined }}
        >
          <MoreOptionsVertical />
        </HvButton>
      }
    >
      <div style={{ width: "500px", height: "500px", background: theme.palette.atmo2 }} />
    </HvBaseDropdown>
  );
};

Custom.story = {
  parameters: {
    v3: true
  }
};
