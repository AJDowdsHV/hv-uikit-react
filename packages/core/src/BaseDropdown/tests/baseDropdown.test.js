import React from "react";
import { render, fireEvent, waitFor } from "testing-utils";
import userEvent from "@testing-library/user-event";
import { Main } from "../stories/BaseDropdown.stories.test";

describe("BaseDropDown", () => {
  describe("Selection List Snapshot testing", () => {
    it("Selection List Container", () => {
      const { container } = render(<Main />);
      expect(container).toMatchSnapshot();
    });
  });

  it("opens on click", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    userEvent.click(baseDropdownHeader); // open

    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "true");
    });
  });

  it("closes on double click", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    userEvent.click(baseDropdownHeader); // open
    userEvent.click(baseDropdownHeader); // close

    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");
    });
  });

  it("opens on Enter", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    fireEvent.keyDown(baseDropdownHeader, { key: "Enter", keyCode: 13 });
    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "true");
    });
  });

  it("closes on double Enter", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    fireEvent.keyDown(baseDropdownHeader, { key: "Enter", keyCode: 13 });
    fireEvent.keyDown(baseDropdownHeader, { key: "Enter", keyCode: 13 });

    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");
    });
  });

  it("opens on Space", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    fireEvent.keyDown(baseDropdownHeader, { key: " ", keyCode: 32 });

    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "true");
    });
  });

  it("closes on double Space", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    fireEvent.keyDown(baseDropdownHeader, { key: " ", keyCode: 32 });
    fireEvent.keyDown(baseDropdownHeader, { key: " ", keyCode: 32 });

    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");
    });
  });

  it("opens and closes mixing mouse click, Enter, and Space", async () => {
    const { getByRole } = render(<Main />);

    const baseDropdownHeader = getByRole("combobox");
    expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");

    userEvent.click(baseDropdownHeader); // open
    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "true");
    });

    fireEvent.keyDown(baseDropdownHeader, { key: "Enter", keyCode: 13 }); // close
    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");
    });

    fireEvent.keyDown(baseDropdownHeader, { key: " ", keyCode: 32 }); // open
    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "true");
    });

    userEvent.click(baseDropdownHeader); // open
    await waitFor(() => {
      expect(baseDropdownHeader).toHaveAttribute("aria-expanded", "false");
    });
  });
});
