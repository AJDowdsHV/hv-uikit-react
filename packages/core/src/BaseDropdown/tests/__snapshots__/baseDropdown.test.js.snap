// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDropDown Selection List Snapshot testing Selection List Container 1`] = `
<div>
  <div
    style="width: 121px;"
  >
    <div
      class="HvBaseDropdown-root-1"
    >
      <div
        aria-expanded="false"
        aria-label="Main sample"
        class="HvBaseDropdown-anchor-2"
        role="combobox"
        tabindex="-1"
      >
        <div
          aria-label="Main sample"
          class="HvBaseDropdown-header-6"
          role="textbox"
          tabindex="0"
        >
          <div
            class="HvBaseDropdown-selection-12"
          >
            <p
              class="HvBaseDropdown-placeholder-13 HvTypography-root-21 HvTypography-placeholderText-35 HvTypography-noWrap-42"
            >
              Placeholder...
            </p>
          </div>
          <div
            class="HvBaseDropdown-arrow-11 HvIconBase-root-43 HvIconBase-xs-44"
            name="DropDownXS"
          >
            <svg
              focusable="false"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="color0"
                d="M5.99994 8.65L1.4 4.05l.7-.7L6 7.25l3.9-3.9.7.7z"
                fill="#414141"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
